src/sequence/AnimateableProperty.cpp

```diff
@@ -237,6 +237,28 @@ float AnimateableProperty::readSingle(float position)
 	return value;
 }
 
+float AnimateableProperty::readFrameSingle(int index) const
+{
+	assert(m_count == 1);
+
+	int last = getSize() / (sizeof(float) * m_count) - 1;
+
+	if(index >= last)
+	{
+		index = last;
+	}
+
+	if(index < 0)
+	{
+		index = 0;
+	}
+
+	sample_t* buf = getBuffer();
+
+	return buf[index];
+}
+
+
```

Comment:
Instead of this method, I would add an explicit cast operator to buffer and then access the buffer directly, considering the range/size of the `Buffer` (`explicit operator const Buffer&() const { return *this; }`), otherwise you check these ifs for every sample during seek which is quite inefficient.